src/back/spv/writer.rs

```diff
@@ -1349,11 +1358,31 @@ impl Writer {
                     }
                 };
 
-                let (result_type_id, result_lookup_ty) = if result_side_left {
-                    (left_result_type_id, left_lookup_ty)
+                let is_comparison = match op {
```

Comment:
This can be made for the follow up, but ideally we'd not have 2 matches of the the same thing. We should combine the matches. i.e. this would look like:
```rust
enum BinaryResult {
  Left,
  Right,
  Bool,
}

let (spirv_op, bin_result) = match op {
  ...
};

let (result_type_id, result_lookup_ty) = match bin_result {
...
};
```

Reply:
Agreed, will note it in my todo list and create an issue at a later time.